eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -687,16 +687,21 @@ object Helpers {
 
         val htlcTxs: Map[OutPoint, Option[HtlcTx]] = claimHtlcOutputs(keyManager, commitments)
 
-        // If we don't have pending HTLCs, we don't have funds at risk, so we can aim for a slower confirmation.
-        val confirmCommitBefore = htlcTxs.values.flatten.map(htlcTx => htlcTx.confirmBefore).minOption.getOrElse(currentBlockHeight + feeTargets.commitmentWithoutHtlcsBlockTarget)
-        val claimAnchorTxs: List[ClaimAnchorOutputTx] = List(
-          withTxGenerationLog("local-anchor") {
-            Transactions.makeClaimLocalAnchorOutputTx(tx, localFundingPubKey, confirmCommitBefore)
-          },
-          withTxGenerationLog("remote-anchor") {
-            Transactions.makeClaimRemoteAnchorOutputTx(tx, commitments.remoteParams.fundingPubKey)
-          }
-        ).flatten
+        val skipAnchors = !feeTargets.spendAnchorWithoutHtlcs && htlcTxs.isEmpty
+        val claimAnchorTxs: List[ClaimAnchorOutputTx] = if (!skipAnchors) {
+          // If we don't have pending HTLCs, we don't have funds at risk, so we can aim for a slower confirmation.
+          val confirmCommitBefore = htlcTxs.values.flatten.map(htlcTx => htlcTx.confirmBefore).minOption.getOrElse(currentBlockHeight + feeTargets.commitmentWithoutHtlcsBlockTarget)
+          List(
+            withTxGenerationLog("local-anchor") {
+              Transactions.makeClaimLocalAnchorOutputTx(tx, localFundingPubKey, confirmCommitBefore)
+            },
+            withTxGenerationLog("remote-anchor") {
+              Transactions.makeClaimRemoteAnchorOutputTx(tx, commitments.remoteParams.fundingPubKey)
+            }
+          ).flatten
+        } else {
+          Nil
+        }
```

Comment:
Nit:

```suggestion
        val spendAnchors = htlcTxs.nonEmpty || feeTargets.spendAnchorWithoutHtlcs
        val claimAnchorTxs: List[ClaimAnchorOutputTx] = if (spendAnchors) {
          // If we don't have pending HTLCs, we don't have funds at risk, so we can aim for a slower confirmation.
          val confirmCommitBefore = htlcTxs.values.flatten.map(htlcTx => htlcTx.confirmBefore).minOption.getOrElse(currentBlockHeight + feeTargets.commitmentWithoutHtlcsBlockTarget)
          List(
            withTxGenerationLog("local-anchor") {
              Transactions.makeClaimLocalAnchorOutputTx(tx, localFundingPubKey, confirmCommitBefore)
            },
            withTxGenerationLog("remote-anchor") {
              Transactions.makeClaimRemoteAnchorOutputTx(tx, commitments.remoteParams.fundingPubKey)
            }
          ).flatten
        } else {
          Nil
        }
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/2246/commits/82e1f5e01819353dd88114d16bfd32d6f6e03398

Reply:
I'm a positive person 😆